app/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnector.scala

```diff
@@ -89,8 +93,9 @@ class EISConnectorImpl(
 
   def onFailure(response: Either[RoutingError, Unit], retryDetails: RetryDetails): Future[Unit] =
     response.left.toOption.get match {
-      case RoutingError.Upstream(upstreamErrorResponse) => attemptRetry(s"with status code ${upstreamErrorResponse.statusCode}", retryDetails)
-      case RoutingError.Unexpected(message, _)          => attemptRetry(s"with error $message", retryDetails)
+      case RoutingError.Upstream(upstreamErrorResponse)       => attemptRetry(s"with status code ${upstreamErrorResponse.statusCode}", retryDetails)
+      case RoutingError.DuplicateLRNError(message, code, lrn) => attemptRetry(s"with status code $code and error $message", retryDetails)
```

Comment:
We definitely do not want to retry on a duplicate LRN error, because we know that will never succeed.

Reply:
Yes you are right, updated it and added test as well to cover this scenario